tools/cloud-build/daily-tests/ansible_playbooks/tasks/check_stockout.yml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+
+- name: Assert variables are defined
+  ansible.builtin.assert:
+    that:
+    - project is defined
+
+- name: Check compute stockout
+  changed_when: false
+  register: stockout
+  until: stockout.rc == 0
+  retries: 5
+  delay: 10
+  vars:
+    stockout_string: Region does not currently have sufficient capacity for the requested resources.
+  ansible.builtin.command: >-
+    gcloud logging --project {{ project }} read
+    'protoPayload.response.error.errors.message="{{ stockout_string }}" AND protoPayload.request.perInstanceProperties.key=~".*{{ slurm_cluster_name }}.*"'
+    --flatten="protoPayload.request.perInstanceProperties"
+    --format='table(protoPayload.request.perInstanceProperties.key:label=INSTANCE_ID,protoPayload.response.error.errors.message.list():label=ERROR_MESSAGE)'
+    --limit=5
+  when: slurm_cluster_name is defined
```

Comment:
Is this intentional to not put this in assert? My guess is that you are hoping it can be reused for other stockout log searches.

Reply:
Yes, that was my intention.  Additionally, this is just a check at failure time, so there's no reason for it to fail.